example/integration_test/error_handling_test.dart

```diff
@@ -0,0 +1,114 @@
+    });
+    expect(fetchEvaluationsRs.isFailure, true,
+        reason:
+            "fetchEvaluations() should fail ${fetchEvaluationsRs.toString()}");
+    expect(
+        fetchEvaluationsRs.asFailure.exception, isA<BKTIllegalStateException>(),
+        reason:
+            "exception should be BKTIllegalStateException but got ${fetchEvaluationsRs.toString()}");
+
+    BKTResult<void> flushRs = await BKTClient.instance.flush().then((value) {
+      return value;
+    }, onError: (obj, st) {
+      fail("flush() should not throw an exception");
+    });
+    expect(flushRs.isFailure, true, reason: "flush() should fail");
+    expect(
+        fetchEvaluationsRs.asFailure.exception, isA<BKTIllegalStateException>(),
+        reason:
+            "exception should be BKTIllegalStateException but got ${fetchEvaluationsRs.toString()}");
+
+    expect(completer.isCompleted, true,
+        reason: "completer should be completed");
+  });
+
+  testWidgets('initialize BKTClient with invalid API_KEY',
+      (WidgetTester _) async {
+    final config = BKTConfigBuilder()
+        .apiKey("RANDOM_KEY")
+        .apiEndpoint(Constants.apiEndpoint)
+        .debugging(debugging)
+        .eventsMaxQueueSize(Constants.exampleEventMaxQueueSize)
+        .eventsFlushInterval(Constants.exampleEventsFlushInterval)
+        .pollingInterval(Constants.examplePollingInterval)
+        .backgroundPollingInterval(Constants.exampleBackgroundPollingInterval)
+        .appVersion(appVersion)
+        .build();
+    assert(config.featureTag == "");
+    final user = BKTUserBuilder().id(userId).customAttributes({}).build();
+
+    await E2EBKTClient.initializeWithRetryMechanism(
+      config: config,
+      user: user,
+    ).then((instanceResult) {
+      expect(instanceResult.isFailure, true,
+          reason: "initialize() should fail ${instanceResult.toString()}");
+      expect(instanceResult.asFailure.exception, isA<BKTForbiddenException>(),
+          reason:
+              "exception should be BKTForbiddenException but got ${instanceResult.toString()}. The exception could be a BKTTimeoutException, but we don't want it here");
+    }, onError: (obj, st) {
+      fail("initialize() should not throw an exception");
+    });
+
+    await BKTClient.instance.fetchEvaluations().then((fetchEvaluationsRs) {
+      expect(
+          fetchEvaluationsRs.asFailure.exception, isA<BKTForbiddenException>(),
+          reason:
+              "exception should be BKTForbiddenException but got ${fetchEvaluationsRs.toString()}");
+    }, onError: (obj, st) {
+      fail(
+          "fetchEvaluations() should not throw an exception ${obj.toString()}");
+    });
+
+    await BKTClient.instance.flush().then((flushRs) {
+      expect(flushRs.isSuccess, true,
+          reason:
+              "flush() should succeed - because no data to flush after we skipped generate events for 401 & 403 errors");
+      return flushRs;
```

Comment:
These are new changes for the E2E tests: no events should be flushed when the SDK is initialized with an invalid API_KEY.

Reply:
I also moved the error handling test cases to a new file to ensure they are tested with fresh data, avoiding any pending events that could affect the results.